### PR TITLE
Discovery page filtering

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- enable filtering in the discovery page
- apply the waterfall layout to account for differences in the recipe card height from the title length using react masonry 

<what will be done in later PRs and not included here>

## Milestones

<the milestones or stories/features that this works towards>

## Resources

<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>

## Test Plan

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
<div>
    <a href="https://www.loom.com/share/64d1f234505e428ca2b3606d36cc492f">
      <p>Discovery page filtering - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/64d1f234505e428ca2b3606d36cc492f">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/pages/RecipeDiscoveryPage.tsx [hunk 1]
```diff
@@ -67,7 +68,7 @@ const RecipeDiscoveryPage = () => {
         filter: recipeFilter,
         offset: 0,
         numRequested: MAX_RECIPES_TO_DISPLAY,
-      });
+      }) ?? [];
```
[the remark this answers is not in the document]
Comment (Brianna-Flo) [reply frontend/src/pages/RecipeDiscoveryPage.tsx:71]: In the case that the axios response data is undefined, so that it wont crash when mapping over array
